katzen w/ katzenpost v0.0.35 fails to build on windows

```
4639757eef28:/build/katzen# make docker-build-windows 
if ! podman images|grep katzen/debian_base; then \
	podman run --replace --name katzen_debian_base docker.io/golang:bookworm bash -c "echo -e 'deb https://deb.debian.org/debian bookworm main\ndeb https://deb.debian.org/debian bookworm-updates main\ndeb https://deb.debian.org/debian-security bookworm-security main' > /etc/apt/sources.list && cat /etc/apt/sources.list && apt update && apt upgrade -y && apt install -y --no-install-recommends build-essential libgles2 libgles2-mesa-dev libglib2.0-dev libxkbcommon-dev libxkbcommon-x11-dev libglu1-mesa-dev libxcursor-dev libwayland-dev libx11-xcb-dev libvulkan-dev gcc-mingw-w64-x86-64" \
	&& podman commit katzen_debian_base katzen/debian_base \
	&& podman rm katzen_debian_base; \
fi
localhost/katzen/debian_base        latest           fdb97e651a08  6 weeks ago    1.71 GB
podman run --rm -v "/build/katzen":/go/katzen/ -v /build/katzen/cache/go:/go/ -e GOCACHE=/go/cache --workdir /go/katzen katzen/debian_base bash -c 'cd /go/katzen/; GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go build -trimpath -ldflags="-H windowsgui -buildid= -X github.com/katzenpost/katzenpost/core/epochtime.WarpedEpoch=false -X github.com/katzenpost/katzenpost/server/internal/pki.WarpedEpoch=false -X github.com/katzenpost/katzenpost/minclient/pki.WarpedEpoch=false" -o katzen.exe'
# github.com/katzenpost/katzen
/usr/local/go/pkg/tool/linux_amd64/link: running x86_64-w64-mingw32-gcc failed: exit status 1
/usr/bin/x86_64-w64-mingw32-ld: /tmp/go-link-2469028699/000011.o: in function `fillrandom_1024_custom':
//_/_/codeberg.org/vula/highctidh@v1.0.2024052900/src/ctidh1024/golang-misc.c:21: multiple definition of `.weak.fillrandom_1024_custom.'; /tmp/go-link-2469028699/000001.o:/tmp/go-build/common.cgo2.c:21: first defined here
/usr/bin/x86_64-w64-mingw32-ld: /tmp/go-link-2469028699/000036.o: in function `fillrandom_2048_custom':
//_/_/codeberg.org/vula/highctidh@v1.0.2024052900/src/ctidh2048/golang-misc.c:21: multiple definition of `.weak.fillrandom_2048_custom.'; /tmp/go-link-2469028699/000026.o:/tmp/go-build/common.cgo2.c:21: first defined here
/usr/bin/x86_64-w64-mingw32-ld: /tmp/go-link-2469028699/000061.o: in function `fillrandom_511_custom':
//_/_/codeberg.org/vula/highctidh@v1.0.2024052900/src/ctidh511/golang-misc.c:21: multiple definition of `.weak.fillrandom_511_custom.'; /tmp/go-link-2469028699/000051.o:/tmp/go-build/common.cgo2.c:21: first defined here
/usr/bin/x86_64-w64-mingw32-ld: /tmp/go-link-2469028699/000086.o: in function `fillrandom_512_custom':
//_/_/codeberg.org/vula/highctidh@v1.0.2024052900/src/ctidh512/golang-misc.c:21: multiple definition of `.weak.fillrandom_512_custom.'; /tmp/go-link-2469028699/000076.o:/tmp/go-build/common.cgo2.c:21: first defined here
collect2: error: ld returned 1 exit status

make: *** [Makefile:30: docker-build-windows] Error 1
```

This works on linux
```
4639757eef28:/build/katzen# make docker-build-linux 
if ! podman images|grep katzen/debian_base; then \
	podman run --replace --name katzen_debian_base docker.io/golang:bookworm bash -c "echo -e 'deb https://deb.debian.org/debian bookworm main\ndeb https://deb.debian.org/debian bookworm-updates main\ndeb https://deb.debian.org/debian-security bookworm-security main' > /etc/apt/sources.list && cat /etc/apt/sources.list && apt update && apt upgrade -y && apt install -y --no-install-recommends build-essential libgles2 libgles2-mesa-dev libglib2.0-dev libxkbcommon-dev libxkbcommon-x11-dev libglu1-mesa-dev libxcursor-dev libwayland-dev libx11-xcb-dev libvulkan-dev gcc-mingw-w64-x86-64" \
	&& podman commit katzen_debian_base katzen/debian_base \
	&& podman rm katzen_debian_base; \
fi
localhost/katzen/debian_base        latest           fdb97e651a08  6 weeks ago    1.71 GB
podman run --rm -v "/build/katzen":/go/katzen/ -v /build/katzen/cache/go:/go/ -e GOCACHE=/go/cache --workdir /go/katzen katzen/debian_base go build -trimpath -ldflags="-buildid= -X github.com/katzenpost/katzenpost/core/epochtime.WarpedEpoch=false -X github.com/katzenpost/katzenpost/server/internal/pki.WarpedEpoch=false -X github.com/katzenpost/katzenpost/minclient/pki.WarpedEpoch=false"
4639757eef28:/build/katzen#
```

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ docker-build-windows: docker-debian-base
 		echo "can only docker-build-windows on debian"; \
 		false; \
 	fi
-	$(docker) $(docker_run_cmd) katzen/$(distro)_base bash -c 'cd /go/katzen/; GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go build -trimpath -ldflags="-H windowsgui ${ldflags}" -o katzen.exe'
+	$(docker) $(docker_run_cmd) katzen/$(distro)_base bash -c 'cd /go/katzen/; HIGHCTIDH_PORTABLE=1 GOOS=windows CGO_ENABLED=1 CGO_LDFLAGS="-Wl,--no-as-needed -Wl,-allow-multiple-definition" CC=x86_64-w64-mingw32-gcc go build -trimpath -ldflags="-H windowsgui ${ldflags}" -o katzen.exe'
 
 docker-android-base:
 	if ! $(docker) images|grep katzen/android_sdk; then \

